server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -61,13 +66,24 @@ export class AssetController {
         const savedAsset = await this.assetService.createUserAsset(authUser, assetInfo, file.path, file.mimetype);
 
```

Comment:
This function now only handles saving the new asset info into the data (thumbnail path if available) and then putting a job into the `asset-uploaded` queue. The first stage in the diagram.

![immich-queue-layour copy](https://user-images.githubusercontent.com/27055614/172532372-f555e953-2b50-4245-91ab-450f728edce7.png)
m